### DISCUSSION TITLE
DietPi-Software | Yacht: Container Management UI

### DISCUSSION THREAD

**Discussion by SelfhostedPro** [MichaIng/DietPi#6269] [Software Request]
Creating a software request
Formal software information

* Software title | Yacht
* Short description | A web interface for managing docker containers with an emphasis on templating to provide 1 click deployments. Think of it like a decentralized app store for servers that anyone can make packages for. 
* Official URL (if available) | https://yacht.sh
* GitHub URL (if available) | https://github.com/SelfhostedPro/Yacht
    (Official) install documentation | https://yacht.sh/docs/Installation/Install (Additional RPi info here: https://yacht.sh/docs/Installation/ARM)

#### Are there similar/alternative software titles available with DietPi-Software?
Portainer

#### What makes your requested software better than the above solutions, if available?
Better docker-compose support (using docker-compose instead of libcompose), more templating options/multiple templates, theming, and a dashboard so you can see the resource usage of all containers at a glance.

#### How can DietPi make the installation easier or compatible, than following the install instructions or do APT installation, if available?

Install with:

```sh
docker volume create yacht_data
docker run -d -p 8000:8000 --name yacht --restart always -v /run/docker.sock:/run/docker.sock -v yacht_data:/config selfhostedpro/yacht
```
Additional theming/whitelabel support with `-E LOGO=<url for logo image>` and `-E THEME=<theme name> ` List of themes [here](https://github.com/SelfhostedPro/Yacht/blob/master/frontend/src/plugins/vuetify.js). If you would like to provide colors for a custom diet-pi theme I would happily add it in for you, just open an issue and I'll get it setup ASAP.

#### Can you provide the installation steps that you would suggest DietPi-Software to do?
Install docker if it's not installed already.

#### Are you willing to help maintaining the software installation, e.g. in case of needed setup changes due to updates etc.? Yes, I could provide a PR if you provide me with the software id you'd like me to use. I'd just be copying this for the most part: https://github.com/MichaIng/DietPi/pull/3933

**Comment by Joulinar** [comment]
Hi,

many thanks for your message. That looks similar to Portainer 😃

**Comment by SelfhostedPro** [comment]
Yeah, it's similar but more focused on small businesses/selfhosting than on the enterprise side of things. It should be very similar as far as adding it to DietPi. If there's anything I can do on my end please feel free to let me know. I'm also happy to see if there are any integrations the community would like added as I'm the developer.

**Comment by jodfie** [comment]
i really would like this added!

**Comment by uniphonic** [comment]
I'd love to see Yacht added to DietPi too! 

I also wonder though, how Yacht would run as a Docker container through Portainer onto DietPi? Even if that's a viable option though, I still would love to see Yacht available directly in DietPi.

Thanks!

**Comment by Joulinar** [comment]
> I also wonder though, how Yacht would run as a Docker container through Portainer onto DietPi?

Basically Portainer as well as Yacht are Docker container itself. Both are somehow equal as they allow managing docker container via a web interface. Portainer is not running any container by it's own. It's just a tool to manage the docker engine.

**Comment by madmedix** [comment]
the beginning of the end for Portainer. The inevitable grab has started with castrating the ce version and obnoxious nagware plowed into the code. Yacht please.

**Comment by MichaIng** [comment]
Converted to discussion. Please vote for this via ⬆️ button at first post.

**Comment by MichaIng** [comment]
This is sitting around too long. We were just talking about it, nice to have an alternative to Portainer. Free software ID would be 177. Generally, can be obtained via `dietpi-software free`, when using `dev` branch.

Installation code form Portainer could be mostly cloned, I guess.

We tested it a little, and at least while Portainer is running as well, some things did not work, like viewing container stats (like CPU usage), templates, images and volumes. It shows a loading bar that persists.

Also no commits since 6 months, I hope it is still developed?

**Reply by SelfhostedPro** [reply]
Active development is over at https://github.com/SelfhostedPro/yacht-nuxt but will be merged back into that repo when complete. If you're good waiting until the next version is released that would be preferable as the current one has some known issues (all related to usage, none to security at least).

**Reply by SelfhostedPro** [reply]
There are instructions for ARM specific installs for stats. The `devel` image has a lot of fixes in it which may be helpful to use instead of latest/main.

**Reply by MichaIng** [reply]
Ah great. If you are doing a larger rework anyway, then its probably better to wait for that. Once it is ready enough for a test implementation, I am open for that. So you could get some early access feedback, and we just need to make clear that it is no stable release yet.
